### PR TITLE
gio: do not take args in application.run()

### PR DESCRIPTION
**Pull request by pbor**
Passing the args is a C-ism of the API since in C the args are
passed to main(), but in rust they are in std::env, so we can
keep the common case simple.
A run_with_args variant is added for the cases where a caller
wants to process the arg vector before running the app.

### CODE REVIEW

File: gio/src/application.rs [hunk 1]
```diff
@@ -12,12 +12,18 @@ use std::mem::transmute;
 
 pub trait ApplicationExtManual {
     #[doc(alias = "g_application_run")]
-    fn run(&self, argv: &[String]) -> i32;
+    fn run(&self) -> i32;
+    #[doc(alias = "g_application_run")]
+    fn run_with_args(&self, argv: &[String]) -> i32;
```
Comment (sdroege) [inline gio/src/application.rs:17]: Ugh, this is annoying :)

```suggestion
    fn run_with_args<S: AsRef<str>>(&self, argv: &[S]) -> i32;
```

please

**Review by sdroege — COMMENTED** [review]
Makes sense to me otherwise, thanks!

**Review by sdroege — APPROVED** [review]

**Review by sdroege — APPROVED** [review]

File: gio/src/application.rs [hunk 2]
```diff
@@ -12,12 +12,19 @@ use std::mem::transmute;
 
 pub trait ApplicationExtManual {
     #[doc(alias = "g_application_run")]
-    fn run(&self, argv: &[String]) -> i32;
+    fn run(&self) -> i32;
+    #[doc(alias = "g_application_run")]
+    fn run_with_args<S: AsRef<str>>(&self, args: &[S]) -> i32;
```
Comment (GuillaumeGomez) [inline gio/src/application.rs:17]: I'm not a big fan of the naming. What do you think about `run_with_environment_args`?

**Review by GuillaumeGomez — COMMENTED** [review]
This is super cool! Just one issue with the naming but otherwise I strongly approve this PR!

File: gio/src/application.rs (hunk 2 above)
Comment (sdroege) [reply gio/src/application.rs:17]: Instead of plain `run()`? That's a lot of additional typing for the normal case :)

Comment (GuillaumeGomez) [reply gio/src/application.rs:17]: I made my comment on `run_with_args`. :stuck_out_tongue:

Comment (sdroege) [reply gio/src/application.rs:17]: It's not using the environment args, but the ones you pass in.

Comment (GuillaumeGomez) [reply gio/src/application.rs:17]: Oh indeed, just realized that it was using argc and argv, not the env. My bad!

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**